mautic/mautic
# Asynchronously loaded segment contact list

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 



#### Description:

This is a performance enhancement. Loading the list of contacts to the tab in the segment detail page is a slow operation and it's slowing down loading the whole segment detail page. This change will let the segment detail page load without it and load the list of contacts asynchronously so it's not blocking the user.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open a segment that has more than 10 contacts in it or create one.
3. Ensure that the contacts tab in the segment detail page gets populated with the correct contacts and the pagination works too.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11005"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
